### DISCUSSION TITLE
zstd: Flush vs Close

### DISCUSSION THREAD

**Discussion by fhkong** [klauspost/compress#984] [Q&A] [answered]
After the encoder has finished writing data, the data flushed to the cache using `Flush` and `Close` functions is different. 
Moreover, the data flushed into the cache using `Flush` function cannot be decoded.

problem reproduce:
```Go
var buf bytes.Buffer
encoder, err := zstd.NewWriter(&buf)
require.NoError(t, err)
encoder.Write(rawData)
encoder.Flush()
fmt.Printf("use Flush compressed data len: %d bytes\n", len(buf.Bytes()))

buf.Reset()
encoder.Reset(&buf)
encoder.Write(rawData)
encoder.Close()
fmt.Printf("use Close compressed data len: %d bytes\n", len(buf.Bytes()))
```

**Comment by klauspost** [comment, marked as answer]
Flush does not indicate EOF, but it will allow the reader to read *up until* where the Flush was used. Therefore a stream ending with a flush is not complete.

Unless you have very special requirements you do not need to call Flush. You will know when you need it.

**Reply by fhkong** [reply]
Okay, thanks
